pkg/skaffold/kubernetes/exec.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2018 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubernetes
+
+import (
+	"io"
+	"os"
+
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/remotecommand"
+)
+
+//Exec contains all required information to execute a command in a container
+type Exec struct {
```

Comment:
Is it possible to reuse [Run](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/exec/exec.go#L268) to execute commands in the container and use the [ExecOptions](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/exec/exec.go#L137) type, instead of creating a new `Exec` type?

Reply:
I tried that before but didn't get the RESTConfig right and tried to do it that way. Though I did some more debugging in the `kubectl` code and maybe found a way to get the rest config to work. I'll push another version tomorrow where I'll use the `ExecOptions`.

Reply:
So I did manage to get everything to work now. Seems like the k8s api has changed much since 1.11. I'd consider upgrading all k8s dependencies to 1.12 as there have been many significant and good changes to the api...